Presentation.Web/Tests/Helpers/ReferenceHelper.ts

```diff
@@ -1,45 +1,34 @@
 ﻿import RefePage = require("../PageObjects/It-system/Tabs/ItSystemReference.po");
 import WaitTimers = require("../Utility/WaitTimers");
 import CSSLocator = require("../object-wrappers/CSSLocatorHelper");
+import ItSystemCatalogPage = require("../PageObjects/it-system/Catalog/ItSystemCatalog.po");
 
 class ReferenceHelper {
     private homePage = new RefePage();
+    private itSystemCatalogPage = new ItSystemCatalogPage();
     private waitUpTo = new WaitTimers();
     private headerButtons = new RefePage().kendoToolbarWrapper.headerButtons();
     private inputFields = new RefePage().kendoToolbarWrapper.inputFields();
     private cssLocator = new CSSLocator();
 
     public createReference(title: string, url: string, id: string) {
-        return this.homePage.getPage()
-            .then(() => browser.wait(this.homePage.isCreateReferenceLoaded(), this.waitUpTo.twentySeconds))
-            .then(() => this.headerButtons.createReference.click())
+        return this.headerButtons.createReference.click()
             .then(() => browser.wait(this.homePage.isReferenceCreateFormLoaded(), this.waitUpTo.twentySeconds))
             .then(() => this.inputFields.referenceDocId.sendKeys(id))
             .then(() => this.inputFields.referenceDocTitle.sendKeys(title))
             .then(() => this.inputFields.referenceDocUrl.sendKeys(url))
             .then(() => this.headerButtons.editSaveReference.click());
     }
 
-    public deleteReference(id: string) {
-        return this.homePage.getPage()
-            .then(() => browser.wait(this.homePage.isCreateReferenceLoaded(), this.waitUpTo.twentySeconds))
-            .then(() => {
-                element.all(by.id("mainGrid")).all(by.tagName("tr")).each((ele) => {
-                    ele.all(by.tagName("td")).each((tdele) => {
+    public goToSpecificItSystemReferences(name: string) {
```

Comment:
Er det med vilje at deleteReference() metoden er fjernet, og at goToSpecificItSystemReferences() ikke bare er tilføjet?

Reply:
Delete function slettes da det ikke er nødvendig at gå så dybt ned i promises for at finde den rigtige knap.
Denne function er erstattet af en simpel xpath function til at finde knappen ud af et reference navn.